test/integration/detachable_joint.cc

```diff
@@ -213,3 +213,145 @@ TEST_F(DetachableJointTest, IGN_UTILS_TEST_DISABLED_ON_WIN32(LinksInSameModel))
+ // is the default offset. Then, linear velocity of 1.0 is published on the
+ // `/cmd_vel` topic. After 200 iterations, the B1 model will remain in the same
+ // position whereas the vehicle will move in the x-direction. Now the
+ // distance between B1 and the vehicle will be the default offset (1.5)
+ // in addition to the distance traveled by the vehicle. Next, B1 is re-attached
+ // to the vehicle. After 200 iterations, we can confirm that B1 has moved with
+ // the vehicle and the distance traveled by B1 is close to that of the vehicle.
+ // Therefore, it confirms that B1 is re-attached to the vehicle.
+ TEST_F(DetachableJointTest, IGN_UTILS_TEST_DISABLED_ON_WIN32(ReAttach))
+ {
+   using namespace std::chrono_literals;
+
+   this->StartServer(common::joinPaths("/test", "worlds",
+       "detachable_joint.sdf"));
+
+   // A lambda that takes a model name and a mutable reference to a vector of
+   // poses and returns another lambda that can be passed to
+   // `Relay::OnPostUpdate`.
+   auto poseRecorder =
+       [](const std::string &_modelName, std::vector<math::Pose3d> &_poses)
+   {
+     return [&, _modelName](const gazebo::UpdateInfo &,
+                            const gazebo::EntityComponentManager &_ecm)
+     {
+       _ecm.Each<components::Model, components::Name, components::Pose>(
+           [&](const Entity &_entity, const components::Model *,
+               const components::Name *_name,
+               const components::Pose *_pose) -> bool
+           {
+             if (_name->Data() == _modelName)
+             {
+               EXPECT_NE(kNullEntity, _entity);
+               _poses.push_back(_pose->Data());
+             }
+             return true;
+           });
+     };
+   };
+
+   std::vector<math::Pose3d> b1Poses, vehiclePoses;
+   test::Relay testSystem1;
+   testSystem1.OnPostUpdate(poseRecorder("B1", b1Poses));
+   test::Relay testSystem2;
+   testSystem2.OnPostUpdate(poseRecorder("vehicle_blue", vehiclePoses));
+
+   this->server->AddSystem(testSystem1.systemPtr);
+   this->server->AddSystem(testSystem2.systemPtr);
+
+   transport::Node node;
+   // time required for the child and parent links to be attached
+   igndbg << "Initially attaching the links" << std::endl;
+   const std::size_t nItersInitialize{100};
+   this->server->Run(true, nItersInitialize, false);
+
+   // detach the B1 model from the vehicle model
+   auto pub = node.Advertise<msgs::Empty>("/B1/detach");
+   pub.Publish(msgs::Empty());
+   std::this_thread::sleep_for(250ms);
+   const std::size_t nItersAfterDetach{100};
+   this->server->Run(true, nItersAfterDetach, false);
+
+   ASSERT_EQ(nItersAfterDetach + nItersInitialize, b1Poses.size());
+   ASSERT_EQ(nItersAfterDetach + nItersInitialize, vehiclePoses.size());
+
+   // Deafult distance between B1 and the vehicle is 1.5.
+   auto defaultDist = 1.5;
+   // Although detached, distance (x-axis) between B1 and vehicle should be 1.5.
+   EXPECT_TRUE(abs(vehiclePoses.back().Pos().X() -
```

Comment:
nit: I think it's better to do `EXPECT_NEAR(vehiclePoses.back().Pose.X(), b1Poses.back().Pos().X() - defaultDist, 0.0001)` because it has a better error message when the expectation fails.

Reply:
addressed [here](https://github.com/gazebosim/gz-sim/pull/1687/commits/4154296935bea0590eba0e61795cd0d0372e8e69#diff-62dd49d28ff49e56afa8f3787907a979acee6497f90d001fe1656e71004a5ccdR288-R289)